test_zeroconf.py

```diff
@@ -807,6 +807,112 @@ def remove_service(self, zeroconf, type, name):
             zeroconf_browser.close()
 
+    def test_integration_with_listener_class_with_update(self):
+
+        service_added = Event()
+        service_removed = Event()
+        service_updated = Event()
+
+        subtype_name = "My special Subtype"
+        type_ = "_http._tcp.local."
+        subtype = subtype_name + "._sub." + type_
+        name = "xxxyyyæøå"
+        registration_name = "%s.%s" % (name, subtype)
+
+        class MyListener(r.ServiceListener):
+            def add_service(self, zeroconf, type, name):
+                zeroconf.get_service_info(type, name)
+                service_added.set()
+
+            def remove_service(self, zeroconf, type, name):
+                service_removed.set()
+
+        class MySubListener(r.ServiceListener):
+            def add_service(self, zeroconf, type, name):
+                pass
+
+            def remove_service(self, zeroconf, type, name):
+                pass
+
+            def update_service(self, zeroconf, type, name):
+                service_updated.set()
+
+        listener = MyListener()
+        zeroconf_browser = Zeroconf(interfaces=['127.0.0.1'])
+        zeroconf_browser.add_service_listener(subtype, listener)
+
+        properties = dict(
+            prop_none=None,
+            prop_string=b'a_prop',
+            prop_float=1.0,
+            prop_blank=b'a blanked string',
+            prop_true=1,
+            prop_false=0,
+        )
+
+        zeroconf_registrar = Zeroconf(interfaces=['127.0.0.1'])
+        desc = {'path': '/~paulsm/'}  # type: r.ServicePropertiesType
+        desc.update(properties)
+        info_service = ServiceInfo(
+            subtype, registration_name,
+            socket.inet_aton("10.0.1.2"), 80, 0, 0,
+            desc, "ash-2.local.")
+        zeroconf_registrar.register_service(info_service)
+
+        try:
+            service_added.wait(1)
+            assert service_added.is_set()
+
+            # short pause to allow multicast timers to expire
+            time.sleep(3)
+
+            # clear the answer cache to force query
+            for record in zeroconf_browser.cache.entries():
+                zeroconf_browser.cache.remove(record)
+
+            # get service info without answer cache
+            info = zeroconf_browser.get_service_info(type_, registration_name)
+            assert info is not None
+            assert info.properties[b'prop_none'] is False
+            assert info.properties[b'prop_string'] == properties['prop_string']
+            assert info.properties[b'prop_float'] is False
+            assert info.properties[b'prop_blank'] == properties['prop_blank']
+            assert info.properties[b'prop_true'] is True
+            assert info.properties[b'prop_false'] is False
+
+            info = zeroconf_browser.get_service_info(subtype, registration_name)
+            assert info is not None
+            assert info.properties[b'prop_none'] is False
+
```

Comment:
Can you remove all the unnecessary (already verified in `test_integration_with_listener_class `) assertions above? Or actually maybe move the new code to the `test_integration_with_listener_class` method so that we don't have that much code repeated.

Reply:
Yes, I will refactor as you suggest